### PR TITLE
Bug: Flow animation kicks in if the edit button is clicked

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #343

## Summary

Bug: Flow animation kicks in if the edit button is clicked on the BasemapLayerList jsapi widget #343

- Adds fix to only show animation if the flow count has actually changed.
- Refactors frame animation class for improved readability.

### CODE REVIEW

File: src/components/calcite-flow/calcite-flow.tsx [hunk 1]
```diff
@@ -72,6 +72,24 @@ export class CalciteFlow {
   //
   // --------------------------------------------------------------------------
 
+  getFlowDirection = (oldFlowCount: number, newFlowCount: number): FlowDirection => {
```
Comment (jcfranco) [inline src/components/calcite-flow/calcite-flow.tsx]: [Nitpick] Maybe include `null` in the return type: `FlowDirection | null`.

File: src/components/calcite-flow/calcite-flow.tsx [hunk 2]
```diff
@@ -72,6 +72,24 @@ export class CalciteFlow {
...
+    const flowCountChanged = oldFlowCount !== newFlowCount;
+
+    if (!flowCountChanged) {
+      return null;
+    }
+
+    const prevHasMulti = oldFlowCount > 1;
```
Comment (jcfranco) [inline src/components/calcite-flow/calcite-flow.tsx]: [Nitpick] Using mixed terminology for old/new, prev/curr.

Comment (jcfranco) [reply src/components/calcite-flow/calcite-flow.tsx]: [Nitpick] What is the `Multi` suffix for? `MultipleFlows`? `MultiThreshold`?

**Comment by driskull** [comment]
Addressed nitpicks

**Pull request merged**